tests/src/whisk/core/apigw/actions/test/ApiGwRoutemgmtActionTests.scala

```diff
@@ -0,0 +1,470 @@
+import whisk.core.WhiskConfig
+import whisk.core.entity.Subject
+import whisk.core.entity.WhiskEntityStore
+import whisk.core.entity.WhiskAuth
+import whisk.core.entity.WhiskAuthStore
+
+case class ApiAction(
+    name: String,
+    namespace: String,
+    backendMethod: String = "POST",
+    backendUrl: String,
+    authkey: String) {
+    def toJson(): JsObject = {
+        return JsObject(
+            "name" -> name.toJson,
+            "namespace" -> namespace.toJson,
+            "backendMethod" -> backendMethod.toJson,
+            "backendUrl" -> backendUrl.toJson,
+            "authkey" -> authkey.toJson
+        )
+    }
+}
+
+/**
+ * Tests for basic CLI usage. Some of these tests require a deployed backend.
+ */
+@RunWith(classOf[JUnitRunner])
+class ApiGwRoutemgmtActionTests
+    extends TestHelpers
+    with BeforeAndAfterAll
+    with WskActorSystem
+    with WskTestHelpers
+    with JsHelpers {
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk
+
+    // Use the whisk.system authentication id and pwd for invoking whisk.system private actions
+    val config = new WhiskConfig(WhiskAuthStore.requiredProperties ++ WhiskEntityStore.requiredProperties)
+    val authstore = WhiskAuthStore.datastore(config)
+    val whiskkey = Await.result(WhiskAuth.get(authstore, Subject("whisk.system"),false)(TransactionId.unknown), 5.seconds)
+    val whisksystemAuthId = whiskkey.authkey.uuid.toString
+    val whisksystemAuthPwd = whiskkey.authkey.key.toString
+
+    override def afterAll() {
+        println("Shutting down authstore connection")
+        authstore.shutdown()
+        super.afterAll()
+    }
+
+    def getApis(
+               bpOrName: String,
+               relpath: String = "",
+               operation: String = "",
+               docid: String = ""): JsArray = {
+        val parms = Map[String,JsValue]() ++
+            Map("namespace" -> "_".toJson) ++
+            { if (bpOrName.length() > 0)  Map("basepath" -> bpOrName.toJson) else Map[String,JsValue]() } ++
+            { if (relpath.length() > 0)   Map("relpath" -> relpath.toJson) else Map[String,JsValue]() } ++
+            { if (operation.length() > 0) Map("operation" -> operation.toJson) else Map[String,JsValue]() } ++
+            { if (docid.length() > 0)     Map("docid" -> docid.toJson) else Map[String,JsValue]() }
+        val wskprops = new WskProps(authKey = whisksystemAuthId+":"+whisksystemAuthPwd)
+        val rr = wsk.action.invoke(
+            name = "routemgmt/getApi",
+            parameters = parms,
+            blocking = true,
+            result = true,
+            expectedExitCode = SUCCESS_EXIT)(wskprops)
+        var apiJsArray: JsArray =
+            try {
+                var apisobj = rr.stdout.parseJson.asJsObject.fields("apis")
+                apisobj.convertTo[JsArray]
+            } catch {
+                case e:Exception =>
+                    JsArray.empty
+            }
+        return apiJsArray
```

Comment:
One does not simply use `return` in Scala :).

Above code is equivalent to the more idiomatic: `Try(rr.stdout.parseJson.asJsObject.fields("apis").convertTo[JsArray]).getOrElse(JsArray.empty)`